components/organisms/ToolList/nav-item.tsx

```diff
@@ -13,6 +14,8 @@ interface NavItemProps {
 
 const NavItem: React.FC<NavItemProps> = ({ username, filterName, tool, selectedFilter, selectedTool }) => {
   const [total, setTotal] = useState<number>();
+  const router = useRouter();
+  const { range } = router.query;
```

Comment:
refactor (non-blocking): Set a default value here and don't bother with the `?? "30"` below.

```suggestion
  const { range = "30" } = router.query;
```